### PR TITLE
Generate Authfile for origin server

### PR DESCRIPTION
**Pull request by bbockelm**
This PR adds the ability (via CLI only; web API to be added once the review gets underway) to generate an authfile for an OSG data federation origin server.

For this to work:
- VO must explicitly list the allowed origins and caches for its namespaces.
- Origin server must be registered and list its supported VOs.
- Cache server must be registered, provide a DN it will use to access the origin, and list its supported VOs.

I did some touch-ups to topology data so we have at least one registered, authenticated origin.

```
$ PYTHONPATH=src ./bin/osg-origin-authfile hcc-anvil-175-55.unl.edu
u 06b434e5.0 /hcc/focusday lr
```

Once the web API is implemented, I expect a systemd service on the origin to generate this authfile.

Resolves: https://opensciencegrid.atlassian.net/browse/SOFTWARE-3469

### CODE REVIEW

**Comment by bbockelm** [comment]
Tested the webapp changes via running it locally.  Works appropriately for the HCC and LIGO origin hostnames.

File: src/stashcache.py [hunk 1]
```diff
@@ -140,3 +140,139 @@ def generate_public_authfile(vo_data, legacy=True):
...
+            raise ValueError("VO {} enables StashCache but does not specify the allowed caches.".format(vo_name))
+
+    resources = []
+    for group in resource_groups:
+        for resource in group.resources:
+            # First, does this provide a cache service?
+            if 'XRootD cache server' not in resource.service_names:
+                continue
+
+            # Next, does it allow this VO?  Unlike the StashCache case above,
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply src/stashcache.py]: Referring to the origins requiring `AllowedVOs`.  Improved the comment in the latest commit.

File: src/webapp/topology.py [hunk 2]
```diff
@@ -126,6 +126,11 @@ def get_tree(self, authorized=False, filters: Filters = None) -> MaybeOrderedDic
         elif filters.has_wlcg is True:
             return
 
+        if 'DN' in new_res:
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply src/webapp/topology.py]: Yup - we want to remove these from the generated XML (as it doesn't fit a fixed schema).  Added a comment in the latest commit.

**Comment by bbockelm** [comment]
@brianhlin - from your follow-up questions:

> Can we run this on the ITB instance for a few days before merging?

Certainly!  I admittedly don't know how to do this, however.

If we have documentation on how to update the ITB instance, I'll gladly follow it.  Otherwise, we might need @djw8605 or @matyasselmeci to step in if you don't know either

> And do we need a new JIRA ticket to update the registration docs to explain how to specify public namespaces and accepting any VO?

Yes, please file a JIRA ticket and attach it to the xcache-1.0 epic.  We'll need to tweak the templates 
too and review the language there.

**Comment by brianhlin** [comment]
ITB deployment instructions here: https://opensciencegrid.org//operations/services/topology/#testing-changes-on-the-itb-instance

JIRA ticket here: https://opensciencegrid.atlassian.net/browse/SOFTWARE-3518. Could you take a look and review the description?

File: src/stashcache.py [hunk 3]
```diff
@@ -140,3 +140,139 @@ def generate_public_authfile(vo_data, legacy=True):
...
+            # we do not consider the lack of AllowedVOs an error as the cache doesn't
+            # explicitly record *which* data federation it is participating in (might not be SC!).
+            matches_vo = False
+            for vo in resource.data.get("AllowedVOs", []):
+                if vo == 'ANY':
+                    matches_vo = True
+                    break
+                elif vo == 'PUBLIC':
+                    continue
+                elif vo == vo_name:
```
Comment (matyasselmeci) [inline src/stashcache.py]: `vo_name` is undefined in several places in this function. Where is it supposed to come from?

Comment (matyasselmeci) [reply src/stashcache.py]: I'll fix this in a PR to your PR.

**Pull request merged**